src/master-playlist-controller.js

```diff
@@ -506,6 +506,42 @@ export class MasterPlaylistController extends videojs.EventTarget {
     }
   }
 
+  /**
+   * Re-tune playback quality level for the current player
+   * conditions. This method will perform destructive actions like removing
+   * already buffered content in order to readjust the currently active
+   * playlist quickly. This is good for manual quality changes
+   *
+   * @private
+   */
+  fastQualityChange_() {
+    let media = this.selectPlaylist();
+
+    if (media !== this.masterPlaylistLoader_.media()) {
+      const hls = this.hls_;
+      const sourceBuffer = this.mainSegmentLoader_.sourceUpdater_.sourceBuffer_;
+      const setCurrentTime = this.tech_.setCurrentTime.bind(this.tech_);
+      const currentTime = this.tech_.currentTime();
+
+      this.masterPlaylistLoader_.media(media);
+
+      // delete all buffered data to allow an immediate quality switch
+      this.mainSegmentLoader_.resetEverything();
+
+      // Wait for the buffer to finish clearing, then seek in place to give the
+      // browser a kick to remove any cached frames from the previous rendition
+      sourceBuffer.one('updateend', function() {
```

Comment:
@gesinger I suppose if we are in the middle of an append operation when the quality switch is initiated it might be possible? When we get the `updateend` we  could check the segment loader state to make sure it isn't `'APPENDING'`. Thoughts?

Reply:
@gesinger Makes sense. Went ahead and made those changes so let me know how they look.